Admin/AbstractAdmin.php

```diff
@@ -1625,11 +1625,7 @@ public function getSubject()
     {
         if ($this->subject === null && $this->request) {
             $id = $this->request->get($this->getIdParameter());
-            if (!preg_match('#^[0-9A-Fa-f\-]+$#', $id)) {
-                $this->subject = false;
-            } else {
-                $this->subject = $this->getModelManager()->find($this->class, $id);
-            }
+            $this->subject = $this->getModelManager()->find($this->class, $id);
```

Comment:
From `ModelManagerInterface::find`:

```
@return object the object with id or null if not found
```

So the subject will not be `false` but `null` for some cases. I don't think we should consider it as BC break but it would be great to have a not on the `UPGRADE-3.x.md`.